domain/blockdag/dag.go

```diff
@@ -87,15 +86,19 @@ type BlockDAG struct {
 	notificationsLock sync.RWMutex
 	notifications     []NotificationCallback
 
-	lastFinalityPoint *blockNode
-
 	utxoDiffStore *utxoDiffStore
 	multisetStore *multisetStore
 
 	reachabilityTree *reachabilityTree
 
 	recentBlockProcessingTimestamps []mstime.Time
 	startTime                       mstime.Time
+
+	tips blockSet
+
+	// validTips is a set of blocks with the status "valid", which have no valid descendants.
+	// Note that some validTips might not be actual tips.
```

Comment:
Please write an essay somewhere explaining this. I will definitely forget otherwise.

Reply:
I'm not sure I understand what exactly you want to be written in said essay, and where do you want to see it?

Reply:
```Tips - actual tips
ValidTips - tips of the valid dag (a.k.a. all blocks that have no statusValid descendants. A.k.a. those that have virtual as their DiffChild)
Virtual parents - parents of virtual. Have some complicated rules to determine them (see selectVirtualParents)```